### PR TITLE
V.4 bnc

### LINKED ISSUE #37: Update M12 board with BNC connector

### LINKED ISSUE #38: Investigate removing fan box board with BNC connector

### PR DESCRIPTION
**Pull request by mechatroniks-git**
Finally updating the repo with the design and manufacturing files.

3 Boards have been made (Oshpark) and consumed in the April 10 Papra Build.

v0.5 is imminent, so no more of these should need to be built. 

Removes fan box board, as there is no longer one with the BNC design. Also updates firmware
Closes #37 
Closes #38

### CODE REVIEW

**Review by darraghbr — COMMENTED** [review]
Should we route the mounting hole on the BNC connector to GND?

Have we also got a .CPL file for V4?

We also need a new Fusion drawing for the PCBA.

**Comment by mechatroniks-git** [comment]
> Should we route the mounting hole on the BNC connector to GND? 
Too late, v.4 PCB has been fabbed.
This can add to v.5, but the mounting pins are isolated from each other and isolated from the BNC ground and center pin. So electrically, won't do anything to ground them, so it's just a manufacturing and mechanical decision. 
 
> Have we also got a .CPL file for V4? 
No, we didn't need one as these were hand stuffed. 

> We also need a new Fusion drawing for the PCBA.
Yes, model has been updated in Fusion, will generate some drawings

**Comment by darraghbr** [comment]
Are you sure they are electrically isolated? Most of the ones I've seen the mounting pins are connected to signal ground?

If we are planning to do this as a release should we have a .CPL file and drawing before it gets merged?

File: firmware/papracode/papracode.ino [hunk 1]
```diff
@@ -110,11 +120,11 @@ void setup() {
   pinMode(led2, OUTPUT);
   pinMode(led3, OUTPUT);
   pinMode(led4, OUTPUT);
-  pinMode(PWMPin, OUTPUT);  //PA3 - TCA0 WO3, pin1 on 14-pin parts
-  TCA0.SPLIT.CTRLB=TCA_SPLIT_LCMP0EN_bm|TCA_SPLIT_HCMP0EN_bm; //PWM on WO0 & WO3
-  TCA0.SPLIT.LPER=0xFF; // Count down from 255 on WO0/WO1/WO2
-  TCA0.SPLIT.HPER=0xFF; // Count down from 255 on WO3/WO4/WO5
-  TCA0.SPLIT.CTRLA=TCA_SPLIT_CLKSEL_DIV4_gc|TCA_SPLIT_ENABLE_bm; //enable the timer with prescaler of 4
+  pinMode(PWMPin, OUTPUT);  //PB0 - TCA0 WO0, pin9 on 14-pin parts
```
Comment (darraghbr) [inline firmware/papracode/papracode.ino:123]: Will we have to go back to SPLIT if and when the buzzer is implemented?

**Comment by darraghbr** [comment]
The only minor thing I can spot on the board is that the ground plane is off centre, I assume this is because the board has grown and either the poly that defines the ground plane hasn't grown with it or the ground plane hasn't been recalculated. No need to fix unless it keeps you up at night.

**Comment by mechatroniks-git** [comment]
> Are you sure they are electrically isolated? Most of the ones I've seen the mounting pins are connected to signal ground?
> 
> If we are planning to do this as a release should we have a .CPL file and drawing before it gets merged?

Yes, I beeped it out.

We can, might be busy work as nobody should make this board with new and improved versions behind it. Documenting it for historical purposes and because there are 3 physical boards out in the wild.

**Comment by mechatroniks-git** [comment]
> The only minor thing I can spot on the board is that the ground plane is off centre, I assume this is because the board has grown and either the poly that defines the ground plane hasn't grown with it or the ground plane hasn't been recalculated. No need to fix unless it keeps you up at night.

I saw and fixed that on v.5. Won't change on v.4 because existing boards have this trait and can't be modified.

File: firmware/papracode/papracode.ino (hunk 1 above)
Comment (mechatroniks-git) [reply firmware/papracode/papracode.ino:123]: Good catch, let me look

**Comment by darraghbr** [comment]
Can we get an OSHPARK DRU file committed to this repo?

**Comment by mechatroniks-git** [comment]
https://github.com/OSHPark/OSHPark-Eagle-Tools/raw/master/dru/oshpark-2layer.dru

**Comment by mmroden** [comment]
closes #36 
closes #14

**Comment by mechatroniks-git** [comment]
DRU file committed https://github.com/tetrabiodistributed/PAPRA-PCB/pull/43/commits/6c3a88eb9bbd5fc58a799f0a938b7e4898f1453a

**Comment by mechatroniks-git** [comment]
Conflict resolved, ready to review & merge

**Review by darraghbr — CHANGES_REQUESTED** [review]
Can we get a PCB drawing committed as well? Either in this PR or put it in an issue to be done later.

**Review by darraghbr — APPROVED** [review]
LGTM

**Pull request merged**